Source/Scene/ModelExperimental/ModelExperimentalDrawCommand.js

```diff
@@ -474,9 +549,117 @@ function derive2DCommand(command) {
   return derivedCommand;
 }
 
+/**
+ * Tracks how many silhouettes have been created. This value is used to
+ * assign a reference number to the stencil.
+ *
+ * @type {Number}
+ * @private
+ */
+ModelExperimentalDrawCommand.silhouettesLength = 0;
+
+function deriveSilhouetteModelCommand(command, model) {
+  // Wrap around after exceeding the 8-bit stencil limit.
+  // The reference is unique to each model until this point.
+  const stencilReference =
+    ++ModelExperimentalDrawCommand.silhouettesLength % 255;
+  const silhouetteModelCommand = DrawCommand.shallowClone(command);
+  let renderState = clone(command.renderState, true);
+
+  // Write the reference value into the stencil buffer.
+  renderState.stencilTest = {
+    enabled: true,
+    frontFunction: WebGLConstants.ALWAYS,
+    backFunction: WebGLConstants.ALWAYS,
+    reference: stencilReference,
+    mask: ~0,
+    frontOperation: {
+      fail: WebGLConstants.KEEP,
+      zFail: WebGLConstants.KEEP,
+      zPass: WebGLConstants.REPLACE,
+    },
+    backOperation: {
+      fail: WebGLConstants.KEEP,
+      zFail: WebGLConstants.KEEP,
+      zPass: WebGLConstants.REPLACE,
+    },
+  };
+
+  if (model.isInvisible()) {
+    // When the model is invisible, disable color and depth writes,
+    // but still write into the stencil buffer.
+    renderState.colorMask = {
+      red: false,
+      green: false,
+      blue: false,
+      alpha: false,
+    };
+    renderState.depthMask = false;
+  }
+
+  renderState = RenderState.fromCache(renderState);
+  silhouetteModelCommand.renderState = renderState;
+
+  return silhouetteModelCommand;
+}
+
+function deriveSilhouetteColorCommand(command, model) {
+  // Wrap around after exceeding the 8-bit stencil limit.
+  // The reference is unique to each model until this point.
+  const stencilReference = ModelExperimentalDrawCommand.silhouettesLength % 255;
+  const silhouetteColorCommand = DrawCommand.shallowClone(command);
+  let renderState = clone(command.renderState, true);
+  renderState.depthTest.enabled = true;
+  renderState.cull.enabled = false;
+
+  // Render the silhouette in the translucent pass if the command is translucent
+  // or if the silhouette color is translucent. This accounts for translucent
+  // model color, since ModelColorPipelineStage sets the pass to translucent.
+  const silhouetteTranslucent =
+    command.pass === Pass.TRANSLUCENT || model.silhouetteColor.alpha < 1.0;
+  if (silhouetteTranslucent) {
+    silhouetteColorCommand.pass = Pass.TRANSLUCENT;
+    renderState.depthMask = false;
+    renderState.blending = BlendingState.ALPHA_BLEND;
+  }
+
+  // Only render silhouette if the value in the stencil buffer equals the reference
```

Comment:
wait, isn't this rendering when the stencil buffer does _not_ equal the reference? 

To check my understanding, the reference value shows you the pixels where the original model will appear, but the silhouette is only drawn for pixels that overlap the scaled silhouette model but not the original-sized one, correct?